### DISCUSSION TITLE
ASE calculator OFT without MD

### DISCUSSION THREAD

**Discussion by KrumaKruma** [mir-group/flare#320] [Q&A]
Dear Flare developpers,

I really like the active learning approach you have implemented with this tool and wanted to ask if the OFT approach is only possible in context with an MD. Since I usually run MDs with variable cell shape followed by variable cell shape geometry optimization, it would be most convenient for me if the FLARE ase calculator would not run OFT only with an MD but would in each call for energy and forces (whatever context it is: MD, geom. opt., etc.) decide automatically if a full DFT calculation is needed or the ML potential can be used. 

Thank you for your help in advance.

Cheers,
Marco

**Comment by YuuuXie** [comment]
Hi @KrumaKruma, sorry for the very delayed response, I did not check the Discussions board before. If you have some DFT calculations already, you can put them together as a trajectory and use our offline training scheme, where every frame will be checked by uncertainty whether to add to the GP or skip. 

On the other hand, if you mean you want to replace MD with geometry optimization, that is an interesting point and we haven't really implemented this kind of feature yet. I know ASE's BFGS optimizer is pretty much in the same spirit as its MD engines. If you are still interested, you can take a look at that module and see if it can be aligned with FLARE
